### PR TITLE
TEST/UD: Wait for a packet up to 10 sec.

### LINKED ISSUE #1995: [v1.2.x] failure on ud_mlx5/test_ud_slow_timer.retransmit1/0
```
14:26:33 [----------] 4 tests from ud_mlx5/test_ud_slow_timer
14:26:33 [ RUN      ] ud_mlx5/test_ud_slow_timer.retransmit1/0
14:26:34 /scrap/jenkins/scrap/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/3/contrib/../test/gtest/uct/test_ud_slow_timer.cc:109: Failure
14:26:34 Value of: ucs_frag_list_sn(&ep(m_e2)->rx.ooo_pkts)
14:26:34   Actual: 0
14:26:34 Expected: 1
14:26:34 
[  FAILED  ] ud_mlx5/test_ud_slow_timer.retransmit1/0, where GetParam() = ud_mlx5/mlx5_1:1 (925 ms)
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5149/label=hpc-test-node2,worker=3/console

### PR DESCRIPTION
**Pull request by yosefe**
picked from #1635
Fixes #1995

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6006/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6010/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich @alex-mikheev pls take a look

File: test/gtest/uct/test_ud_slow_timer.cc [hunk 1]
```diff
@@ -44,6 +44,15 @@ class test_ud_slow_timer : public ud_base_test {
     {
         return UCS_ERR_INVALID_PARAM;
     }
+
+    void wait_for_rx_sn(unsigned sn)
+    {
+        ucs_time_t deadline = ucs_get_time() +
+                              ucs_time_from_sec(10) * ucs::test_time_multiplier();
```
Comment (brminich) [inline test/gtest/uct/test_ud_slow_timer.cc]: minor: we have `UCT_TEST_TIMEOUT_IN_SEC` and `DEFAULT_TIMEOUT_SEC` macro in uct_test.h

Comment (yosefe) [reply test/gtest/uct/test_ud_slow_timer.cc]: i cherry picked the fix as is from master..

**Pull request merged**